var/spack/repos/builtin/packages/python/package.py

```diff
@@ -324,6 +332,105 @@ def filter_compilers(self, prefix):
     # ========================================================================
 
     @property
+    def executable(self):
+        """Returns the path to the Python executable, which may vary depending
+        on the version of Python and how it was installed.
+
+        In general, Python 2 comes with ``python`` and ``python2`` executables,
+        while Python 3 only comes with a ``python3`` executable.
+
+        :returns: The full path to the Python executable
+        :rtype: str
+        """
+        # We need to be careful here. If the user is using an externally
+        # installed python, all 3 executables could be in the same directory.
+
+        # Search for `python3` iff using Python 3
+        if (self.spec.satisfies('@3:') and
+                os.path.exists(os.path.join(self.prefix.bin, 'python3'))):
+            exe = 'python3'
+        # Search for `python2` iff using Python 2
+        elif (self.spec.satisfies('@:2') and
+                os.path.exists(os.path.join(self.prefix.bin, 'python2'))):
+            exe = 'python2'
+        # If neither were found, try `python`
+        elif os.path.exists(os.path.join(self.prefix.bin, 'python')):
+            exe = 'python'
+        else:
+            msg = 'Cannot locate python executable in {0}'
+            raise RuntimeError(msg.format(self.prefix.bin))
```

Comment:
By passing the search directory, I'm hoping that users who erroneously add `/usr/bin/` to their `packages.yaml` will see that they actually need to use `/usr`.